testapi.pm

```diff
@@ -99,11 +99,11 @@ sub init {
         $serialdev = "hvc0";
     }
     elsif (check_var('VIRSH_VMM_FAMILY', 'xen') && check_var('VIRSH_VMM_TYPE', 'linux')) {
-        if (check_var('VERSION', '12-SP2')) {
-            $serialdev = "hvc0";
+        if (check_var('VERSION', '12-SP1')) {
+            $serialdev = "xvc0";
         }
         else {
-            $serialdev = "xvc0";
+            $serialdev = "hvc0";
```

Comment:
this code really needs to move to main.pm and set SERIALDEV. If that doesn't work, we need to make it work.